curl-helper.c

```diff
@@ -1568,9 +1610,11 @@ static void handle_##name##suffix(Connection *conn, value option) \
 
 #define SETOPT_FUNCTION(name) SETOPT_FUNCTION_(name,FUNCTION)
 #define SETOPT_FUNCTION2(name) SETOPT_FUNCTION_(name,FUNCTION2)
+#define SETOPT_FUNCTION_BUF(name) SETOPT_FUNCTION_(name,FUNCTION_BUF)
```

Comment:
used only once, better inline

Reply:
I'm not sure what you mean here, could you say more?

This seemed similar to CURLOPT_WRITEFUNCTION2 so I just made it look like that.

Reply:
i mean setopt_function2 is needed in several places, so it makes sense it is a macro, but setopt_function_buf is only used once so can instead call `setopt_function_("WRITE",FUNCTION_BUF)` directly. Less new names in scope - less cognitive load.